### PR TITLE
Removes a ton of static type warnings

### PR DESCRIPTION
**Pull request by spookydonut**
## About the PR
some of these are potential runtimes the rest are just weak typing

## Why's this needed?
makes dreamchecker slightly more usable

### CODE REVIEW

File: _stdlib/code/_macros.dm [hunk 1]
```diff
@@ -64,14 +64,14 @@ var/list/global_spawn_dbg = list()
 #define isspythief(x) (istype(x, /mob/living/carbon/human) && x:mind && x:mind:special_role == "spy_thief")
 
 // Why the separate mask check? NPCs don't use assigned_role and we still wanna play the cluwne-specific sound effects.
-#define iscluwne(x) (istype(x, /mob/living/carbon/human) && ((x:mind && x:mind.assigned_role && x:mind:assigned_role == "Cluwne") || istype(x:wear_mask, /obj/item/clothing/mask/cursedclown_hat)))
+#define iscluwne(x) ((x?.mind?.assigned_role == "Cluwne") || istype(x.wear_mask, /obj/item/clothing/mask/cursedclown_hat)))
 #define ishorse(x) (istype(x, /mob/living/carbon/human) && ((x:mind && x:mind.assigned_role && x:mind:assigned_role == "Horse") || istype(x:wear_mask, /obj/item/clothing/mask/horse_mask/cursed)))
```
[the remark this answers is not in the document]
Comment (pali6) [reply _stdlib/code/_macros.dm:68]: I feel like that should sadly be `?:`.

File: code/WorkInProgress/dialogue/dialogueSystem.dm [hunk 2]
```diff
@@ -3,7 +3,7 @@
 */
 
 var/global/list/globalDialogueFlags = list()
-/client/var/list/dialogueNodePath = list()
+/client/var/list/list/dialogueNodePath = list()
```
[the remark this answers is not in the document]
Comment (pali6) [reply code/WorkInProgress/dialogue/dialogueSystem.dm:6]: I believe it's just saying that it's a list of lists.

File: _stdlib/code/_macros.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (ZeWaka) [reply _stdlib/code/_macros.dm:68]: I mean the line below, ishorse(). Darn you github for not being clear!

[the remark this answers is not in the document]
Comment (spookydonut) [reply _stdlib/code/_macros.dm:68]: ishorse either isnt being used or i didnt see it

[the remark this answers is not in the document]
Comment (spookydonut) [reply _stdlib/code/_macros.dm:68]: Yeah ishorse is only used once on something that is already a human typepath

**Pull request merged**